src/main/java/io/github/jhipster/domain/BaseEntity.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016-2018 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see http://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.jhipster.domain;
+
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+
+/**
+ * Base class of all entities generated by JHipster.
+ */
+@MappedSuperclass
+public abstract class BaseEntity implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Unique id for an entity of a given type. Will be {@code null} for an entity that hasn't been saved to its data
+     * storage.
+     */
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
```

Comment:
This would not work for mongo or Cassandra apps as we generate a different Id type

Reply:
Yes... I was pretty sure something like this would happen. Choices are:
* Do not use a base class for mongo and cassandra
* Use a `BaseJpaEntity` on top of `BaseEntity`
* Use a `BaseJpaEntity` and a `BaseNoqlEntity` or something alike
* Put the `BaseEntity` in the generator instead of here to adapt